src/models/actions/ExecuteFragment.js

```diff
@@ -0,0 +1,54 @@
+const { performance } = require("perf_hooks");
+const { BaseAction } = require("./BaseAction");
+
+class ExecuteFragment extends BaseAction {
+  async eval(context) {
+    const t0 = performance.now();
+
+    if (this.errorFromService) {
+      return {
+        actionReports: [
+          {
+            action: "Execute.Fragment",
+            success: false,
+            shortSummary: this.errorFromService,
+            longSummary: null,
+            time: performance.now() - t0,
+          },
+        ],
+      };
+    }
+    try {
+      // Test variables not defined by the fragment will be undefined when executed
+      // from a parent test unless mapped in the fragments inputs.
+      //
+      // The context will contain all secrets, env vars and upstream context results in
+      // addition to the input overrides from the fragment. Because the context is global
+      // its best to explicity override all test variables used in the fragment for clarity.
+      for (const variable in this.parameters.inputs) {
+        context.set(variable, this.parameters.inputs[variable]);
```

Comment:
I noticed a bug when doing this step here, contex.set only updated the data, but not the safeData which is used for presentation in htttp reports. So in theory, you could pass an updated var in the fragment, but the report may show the old var, but in fact, the new var did work. The previous commits to this PR in Context.js fix that issue.